src/crimson/osd/osd_operations/pg_advance_map.cc

```diff
@@ -133,4 +125,76 @@ seastar::future<> PGAdvanceMap::start()
   });
 }
 
+seastar::future<> PGAdvanceMap::split_pg(
+    std::set<std::pair<spg_t, epoch_t>> split_children_info,
+    cached_map_t next_map)
+{
+  using cached_map_t = OSDMapService::cached_map_t;
+  logger().debug("{}: start", *this);
+
+  for (auto& child_pg_info : split_children_info) {
```

Comment:
Can this be done in parallel? using `coroutine::parallel_for_each`

Reply:
I tried it and I think it won't work with with_blocking_event? 
Not sure if it can handle multiple blocking events at a time since we re-use "this"

` co_await this->template with_blocking_event<PGMap::PGCreationBlockingEvent>`

Reply:
IIRC `with_blocking_event` is only used for tracking of operations so we would be able to tell which ops are blocked and for how long (using the `trigger` lifetime). If I'm correct, we should be ok with parallelism as we would like to track each splitted pg creation to identify blocked creations (or to track ones in progress).

This might be confusing with the pipeline concept where we have four dedicated pipelines: connection, per-shard, common pg and common obc that are used to assure ordering and dependency availability (get_trigger).  These are used by `enter_stage` _only_.

Reply:
note: enter_stage calls with_blocking_event internally so that every stage transition is a tracked event as well.